src/lib.rs

```diff
@@ -85,6 +101,33 @@ impl Default for Pid1Settings {
     }
 }
 
+impl Builder {
+    /// Initializes the log builder with defaults.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Should the crate log to [`std::io::Stderr`]. This can be
+    /// useful to detect whether it is running with PID 1. By default
+    /// it is 'false'.
+    pub fn enable_log(&mut self, enable_log: bool) -> &mut Self {
+        self.log = enable_log;
+        self
+    }
+
+    /// Duration to wait for the child process to exit. By default it
+    /// is 2 seconds.
+    pub fn timeout(&mut self, timeout: Duration) -> &mut Self {
+        self.timeout = timeout;
+        self
+    }
+
+    /// Build `Pid1Settings`.
+    pub fn build(self) -> Pid1Settings {
```

Comment:
I'm not sure we need a distinction between the builder and `Pid1Settings`. What if we had `Pid1Settings` only, it uses a builder interface, and then there's simply a `.launch()` or similarly named method on it in place of the top level `relaunch_if_pid1`? On top of that, we could also provide a helper top-level function (maybe still named `relaunch_if_pid1` that uses default `Pid1Settings` to simplify integration.

Reply:
Yeah, I think that would be a better UX. Thanks!

Reply:
Yeah, I think that will be a better UX. Thanks!